tests/test_core_network.py

```diff
@@ -2404,6 +2404,86 @@ def test_removing_mode_from_links_removes_empty_links():
     assert n.has_link("0")
 
 
+def test_splitting_links_removes_required_mode_from_existing_links():
+    n = Network("epsg:27700")
+    n.add_link("0", 1, 2, attribs={"modes": {"car", "bike"}, "length": 1})
+
+    n.split_links_on_mode("bike")
+
+    assert n.link("0")["modes"] == {"car"}, "The link modes have been incorrectly set"
+
+
+def test_splitting_links_creates_new_links_of_required_mode():
+    n = Network("epsg:27700")
+    n.add_link("0", 1, 2, attribs={"modes": {"car", "bike"}, "length": 1})
+
+    new_links = n.split_links_on_mode("bike")
+
+    assert len(new_links) == 1, "A new link was not generated"
+    new_link = list(new_links)[0]
+    assert n.has_link(new_link), "The new link was not found in the Network"
+    assert n.link(new_link)["modes"] == {"bike"}, "The mode of the new link is wrong"
+
+
+def test_data_retained_when_splitting_links():
+    n = Network("epsg:27700")
+    n.add_link("0", 1, 2, attribs={"modes": {"car", "bike"}, "length": 1, "BIG": "DATA"})
+
+    new_links = n.split_links_on_mode("bike")
+
+    assert len(new_links) == 1, "A new link was not generated"
+    new_link = list(new_links)[0]
+    assert "BIG" in n.link(new_link), "Attribute key was not found in the data saved on the link"
+    assert (
+        n.link(new_link)["BIG"] == "DATA"
+    ), "The new link did not inherit the same attribute value"
+
+
+def test_splitting_links_uses_desired_prefix_for_new_link_ids():
+    n = Network("epsg:27700")
+    n.add_link("0", 1, 2, attribs={"modes": {"car", "bike"}, "length": 1})
+
+    new_links = n.split_links_on_mode("bike", link_id_prefix="HEYO-")
+
+    assert len(new_links) > 0, "No new links were generated"
```

Comment:
Will more than one new link be generated in this case?

Reply:
no just one, I'll change it to a stricter condition and get rid of the loop below, since there's just one link